src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql

```diff
@@ -0,0 +1,301 @@
+//     union ["Condition": "Unspecified dementia, with behavioral disturbance"]
+
+define DementiaConditions:
+  (["Condition"] C where (
+    C.code.coding[0].code.value = 'F03.9' or
+    C.code.coding[0].code.value = 'F01.5' or
+    C.code.coding[0].code.value = 'G31.0' or
+    C.code.coding[0].code.value = 'G20' or
+    C.code.coding[0].code.value = 'F03.91'
+  ))
+
+define HasDementia:
+  Length(DementiaConditions) > 0
+
+
+//   depression condition ("F32.A - Depression, unspecified" and "F33.1 - Major Depressive Disorder, Recurrent, Moderate")
+//   and/or 
+//   anxiety condition ("F41.1 - Generalized Anxiety Disorder" - not present in our 3 test patients...)
+
+define MoodConditions: 
+  ["Condition"] C where (
+    C.code.coding[0].code.value = 'F32.A' or
+    C.code.coding[0].code.value = 'F33.1' or
+    C.code.coding[0].code.value = 'F41.1'
+  )
+
+define HasMoodConditions:
+  Length(MoodConditions) > 0
+
+
+// Naming pattern: "CIRG_MINICOG" is Questionnaire.id
+define PresentQnr_CIRG_MINICOG:
+  if dcw_cog_level_1 is true then {
+    id: 'CIRG-MINICOG',
+    schedule: Scheduled_Timing
+  } else null
+
+
+define PresentQnr_CIRG_ECOG12:
+  if dcw_cog_level_1 is true then {
+    id: 'CIRG-ECOG12',
+    schedule: Scheduled_Timing
+  }
+  else null
+
+define PresentQnr_CIRG_CP_ECOG:
+  if dcw_cog_level_1 is true and dcw_care_partner_present is true then {
+    id: 'CIRG-CP-ECOG',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_SLUMS:
+  if dcw_cog_level_2 is true then {
+    id: 'CIRG-SLUMS',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_ADL_IADL:
+  if dcw_cog_level_2 is true and dcw_care_partner_present is true then {
+    id: 'CIRG-ADL-IADL',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_BEHAV5:
+  if dcw_mood_level_1 is true then {
+    id: 'CIRG-BEHAV5',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_GAD7:
+ // TODO figure out what score this is supposed to be
+ // CIRG-SLUMS score is Normal or Mild Neurocognitive Disorder
+ // dependent on education level? using High School Education criteria here 
+  if dcw_mood_level_2 is true and First(SLUMS_Score) >= 21 then {
+    id: 'CIRG-GAD7',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_GDS:
+  // TODO figure out what score this is supposed to be
+  // CIRG-SLUMS score is Normal or Mild Neurocognitive Disorder
+  // dependent on education level? using High School Education criteria here 
+  if dcw_mood_level_2 is true and First(SLUMS_Score) >= 21 then {
+    id: 'CIRG-GDS',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_C_IDAS:
+  // TODO figure out what score this is supposed to be
+  // CIRG-SLUMS score indicates dementia 
+  if dcw_mood_level_2 is true and 
+     dcw_care_partner_present is true and 
+     First(SLUMS_Score) <= 20 // dependent on education level? using High School Education criteria here
```

Comment:
@sierramatice here, too, I am not certain what SLUMS score is indicative of dementia.